misc/userscripts/qute-pass

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+# Insert login information using pass and a dmenu-provider (e.g. dmenu, rofi -dmenu, ...).
+# A short demonstration can be seen here: https://i.imgur.com/KN3XuZP.gif.
+#
+# The domain of the site has to appear as a segment in the pass path, for example: "github.com/cryzed" or
+# "websites/github.com". How the username and password are determined is freely configurable using the CLI arguments.
+# The login information is inserted by emulating key events using qutebrowser's fake-key command in this manner:
+# [USERNAME]<Tab>[PASSWORD], which is compatible with almost all login forms.
+#
+# Dependencies: tldextract (Python 3 module), pass
+# For issues and feedback please use: https://github.com/cryzed/qutebrowser-userscripts
+#
+# Chris Braun (cryzed), 2017
+
+import argparse
+import enum
+import fnmatch
+import functools
+import os
+import re
+import shlex
+import subprocess
+import sys
+
+import tldextract
+
+PASSWORD_STORE_PATH = os.path.expanduser('~/.password-store')
+DMENU_INVOCATION = 'rofi -dmenu'
+ENCODING = 'UTF-8'
+
+argument_parser = argparse.ArgumentParser()
+argument_parser.add_argument('url', nargs='?', default=os.getenv('QUTE_URL'))
+argument_parser.add_argument('--password-store', '-p', default=PASSWORD_STORE_PATH,
+                             help='Path to your pass password-store')
+argument_parser.add_argument('--username-pattern', '-u', default=r'.*/(.+)',
+                             help='Regular expression that matches the username')
+argument_parser.add_argument('--username-target', '-U', choices=['path', 'secret'], default='path',
+                             help='The target for the username regular expression')
+argument_parser.add_argument('--password-pattern', '-P', default=r'(.*)',
+                             help='Regular expression that matches the password')
+argument_parser.add_argument('--dmenu-invocation', '-d', default='rofi -dmenu',
+                             help='Invocation used to execute a dmenu-provider')
+argument_parser.add_argument('--no-insert-mode', '-n', dest='insert_mode', action='store_false',
+                             help="Don't automatically enter insert mode")
+group = argument_parser.add_mutually_exclusive_group()
+group.add_argument('--username-only', '-uo', action='store_true', help='Only insert username')
+group.add_argument('--password-only', '-po', action='store_true', help='Only insert password')
```

Comment:
I have no idea if two-letter short arguments even work, but if they do, it's still kind of odd IMHO. Maybe just don't have short forms of this?

Reply:
Yes, these work, but removing the short-forms entirely feels inconsistent if the other options have one. I'll change them to usernam**e** and pass**w**ord respectively (or remove all short-forms if you dislike that). Personally I never use them anyways, I put them there for the benefit of people who do.